manifestoes/extensions.rst

```diff
@@ -0,0 +1,1675 @@
+  Furthermore, any code written in the ``Student`` edition is expected
+  to compile with all future versions of GHC for 10 years, meant to
+  echo the expected lifetime of a textbook.
+
+With that out of the way, here are some scenarios that the language editions
+model helps us to address:
+
+* **Stability**. Once we include a feature in a ``Stable`` edition, it
+  will not change. However, that does not mean GHC cannot evolve. If we
+  identify a ``Stable`` feature that nevertheless deserves an upgrade, we
+  can do so, provided we continue to support the old behavior. So
+  type inference around ``FlexibleInstances``, say, might work one way in
+  ``Stable2024`` and differently in ``Stable2027``. If we release a new
+  ``Stable`` edition every 3 years, users have three years to upgrade before
+  we no longer guarantee support. A three-year timeline is long enough that
+  we might imagine contributors writing upgrade tools in that timeframe.
+
+* **Error messages**. Because the language edition describes the user (very
+  coarsely!) we can tailor error messages to be appropriate for them. The
+  ``Student`` editions will not advise ``DataKinds``. The ``Stable`` editions
+  will not advise ``LinearTypes``.
+
+* **Warning evolution**. One challenge in adding features to GHC is to
+  know how to evolve the warning system. Does a new warning get added to
+  ``-Wall``? And what's the deal with ``-Wcompat``?
+
+  Language editions make
+  these questions easier to answer: If we think that a warning should be
+  enabled for users going forward, we can turn it on by default, but only
+  in appropriate language editions, such as ``Experimental`` or ``Latest``.
+
+  For ``-Wcompat``, we add warnings that describe features that will change
+  in the next edition in a given series. So the ``Stable2024`` edition of
+  ``-Wcompat`` will warn about features changing in ``Stable2027``. This
+  set of warnings will likely grow between 2024 and 2027; that's fine.
+  (We do *not* guarantee that all ``Stable`` code remain warning-free, especially
+  with ``-Wcompat`` enabled.) Now if a user is running ``Stable2024`` in
+  2028 with ``-Wcompat``, they won't see warnings about changes due in
+  2030; they'll see just the warnings they need to upgrade to ``Stable2027``.
+
+* **Simplicity**. The goal is that a vast majority of our users will be
+  able to specify a language edition, and that's it. No extensions. No
+  warning flags. This simplifies what a user needs to think about when
+  setting up a Haskell project, removing the paralysis of choice that
+  can reign today.
+
+Proposed specification
+~~~~~~~~~~~~~~~~~~~~~~
+
+-  We refresh the concept of *language
+   edition*. Existing language editions ``Haskell98``, ``Haskell2010``, and
+   ``GHC2021`` will continue to be language editions, to which we
+   add the following:
+
+   * ``Stable2024``
+   * ``Experimental2024``
+   * ``Latest``
+   * ``Student2024``
```

Comment:
This seems like a lot. One of the costs of making language editions into powerful things instead of collections of defaults is that the overhead tends to grow over time. It seems like it would be easier going forwards if we instead built these out of some component flags. For example, we could have:
```
--stable-from=2024
```
Which can be used uniformly to control what stability date you want - then `Stable2024` just provides a default value for that.

Similarly, if we provide `--error-message-abstraction=basic`, then `Student2024` can just be `--stable-from=2024 --error-message-abstraction=basic`.

Reply:
I'm not sure I see the payload of this suggestion. We could break these into components, as you suggest. But how would that reduce complexity / overhead?